src/components/List/ListItem.tsx

```diff
@@ -1,20 +1,30 @@
-import type { FC, PropsWithChildren } from 'react';
+import type { ComponentProps, FC, PropsWithChildren } from 'react';
 import { twMerge } from 'tailwind-merge';
 import { mergeDeep } from '../../helpers/merge-deep';
 import { getTheme } from '../../theme-store';
 import type { DeepPartial } from '../../types';
 
 export interface FlowbiteListItemTheme {
-  base: string;
+  icon: string;
+  withIcon: {
+    on: string;
+    off: string;
+  };
 }
 
 export interface ListItemProps extends PropsWithChildren {
   theme?: DeepPartial<FlowbiteListItemTheme>;
   className?: string;
+  icon?: FC<ComponentProps<'svg'>>;
 }
 
-export const ListItem: FC<ListItemProps> = ({ children, className, theme: customTheme = {} }) => {
-  const theme = mergeDeep(getTheme().listGroup.item, customTheme);
+export const ListItem: FC<ListItemProps> = ({ children, className, theme: customTheme = {}, ...props }) => {
```

Comment:
destructure the `icon` prop here and spread the rest of the `...props` to the `<li>` element

Reply:
Isn't it currently like that??

Reply:
partially, because the `...props` are not spread on the <`<li>` element down below

Reply:
Thanks @SutuSebastian , fixed right now...